runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -692,7 +692,33 @@ TR_J9ServerVM::getStaticReferenceFieldAtAddress(uintptrj_t fieldAddress)
 bool
 TR_J9ServerVM::stackWalkerMaySkipFrames(TR_OpaqueMethodBlock *method, TR_OpaqueClassBlock *clazz)
    {
+   if(!method)
+      {
+      return false;
+      }
+
    JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
+   auto *vmInfo = _compInfoPT->getClientData()->getOrCacheVMInfo(stream);
+   
+   if (vmInfo->_jlrMethodInvoke == ((J9Method *) method))
```

Comment:
If `vmInfo->jlrMethodInvoke` is null you have to send a message to the client to get up updated information. This message is actually JITServer::MessageType::VM_stackWalkerMaySkipFrames, but change it such that it returns the value of `vmInfo->jlrMethodInvoke` as well so that we can update at the server.

Reply:
Thanks. Lets see if I am understanding this correctly. 

This `JITServer::MessageType::VM_stackWalkerMaySkipFrames` message currently returns a `bool` value which is the return value of `stackWalkerMaySkipFrames` function at client. I need to change this `VM_stackWalkerMaySkipFrames` message so that it returns the value of `vmInfo->jlrMethodInvoke`. Shall we change the name of this message then? 

If `vmInfo->jlrMethodInvoke` is NULL, we send a message and once have the new `vmInfo->jlrMethodInvoke ` value on the server, we can cache the value inside `VMInfo`.

Reply:
> we can cache the value inside VMInfo.

Note, that cache will need to be updated if the `java.lang.reflect.Method` class is redefined.

Reply:
> cache will need to be updated if the java.lang.reflect.Method class is redefined.

@DanHeidinga  I looked where `vmInfo->jlrMethodInvoke` is set and there is only one place: in `standardInit()` called from `scarInit(J9JavaVM * vm)` called from `J9VMDllMain(J9JavaVM* vm, IDATA stage, void* reserved)` with a stage of `JCL_INITIALIZED`. Thus, if `java.lang.reflect.Method` class is redefined, don't we have to update `vmInfo->jlrMethodInvoke`  as well?